src/main/java/sx/blah/discord/handle/impl/obj/Guild.java

```diff
@@ -779,6 +778,18 @@ public IEmoji getEmojiByName(String name) {
 	}
 
 	@Override
+	public IEmoji addEmoji(String name, BufferedImage image){
+		if (name.length() < 2 || name.length() > 32 || DiscordUtils.EMOJI_CREATE_NAME.matcher(name).find())
+			throw new DiscordException("Emoji names must be at least 2 characters long and can only contain alphanumeric characters and underscores.");
+
+		DiscordUtils.checkPermissions(client, this, EnumSet.of(Permissions.MANAGE_EMOJIS));
+
+		DataBufferByte data = (DataBufferByte) image.getRaster().getDataBuffer();
+		EmojiObject response = ((DiscordClientImpl) client).REQUESTS.POST.makeRequest(DiscordEndpoints.GUILDS + getID() + "/emojis", new EmojiCreateRequest(name, data.getData()), EmojiObject.class);
```

Comment:
@Techtony96 only if the http response code allows it to reach here: https://github.com/austinv11/Discord4J/blob/dev/src/main/java/sx/blah/discord/api/internal/Requests.java#L328 But I believe it should have a hard fail rather than return null on failure